### PR TITLE
More web options (HTTP version, encoding, transfer body, follow request, range)

### PR DESCRIPTION
**Pull request by SpaghettDev**

### CODE REVIEW

**Comment by matcool** [comment]
bro needs to set http version 💀

File: loader/include/Geode/utils/web.hpp [hunk 1]
```diff
@@ -122,12 +134,20 @@ namespace geode::utils::web {
 
         WebRequest& userAgent(std::string_view name);
 
+        WebRequest& encoding(std::string_view encoding);
```
Comment (matcool) [inline loader/include/Geode/utils/web.hpp:137]: encoding of what? the request or the response? and what are valid values?

File: loader/include/Geode/utils/web.hpp [hunk 2]
```diff
@@ -122,12 +134,20 @@ namespace geode::utils::web {
 
         WebRequest& userAgent(std::string_view name);
 
+        WebRequest& encoding(std::string_view encoding);
+
         WebRequest& timeout(std::chrono::seconds time);
 
+        WebRequest& range(std::pair<std::uint64_t, std::uint64_t> byteRange);
```
Comment (matcool) [inline loader/include/Geode/utils/web.hpp:141]: what is this for

File: loader/include/Geode/utils/web.hpp [hunk 3]
```diff
@@ -122,12 +134,20 @@ namespace geode::utils::web {
...
 
+        WebRequest& encoding(std::string_view encoding);
+
         WebRequest& timeout(std::chrono::seconds time);
 
+        WebRequest& range(std::pair<std::uint64_t, std::uint64_t> byteRange);
+
         WebRequest& certVerification(bool enabled);
+        WebRequest& transferBody(bool enabled);
+        WebRequest& followRequest(bool enabled);
```
Comment (matcool) [inline loader/include/Geode/utils/web.hpp:145]: could this be renamed to `followRedirects` or is that a different thing

**Comment by matcool** [comment]
i think some doc comments would help out a bunch of these cuz theyre kinda ambiguous as to what they do

**Comment by SpaghettDev** [comment]
sure

File: loader/include/Geode/utils/web.hpp (hunk 1 above)
Comment (SpaghettDev) [reply loader/include/Geode/utils/web.hpp:137]: [its the encoding of the response](https://curl.se/libcurl/c/CURLOPT_ACCEPT_ENCODING.html), specifically a download response. as for the valid values, it depends on libcurl, but an example of valid values are br, gzip, deflate. these can be passed to encoding as a comma separated string.

File: loader/include/Geode/utils/web.hpp (hunk 2 above)
Comment (SpaghettDev) [reply loader/include/Geode/utils/web.hpp:141]: the range of bytes to download from a response

File: loader/include/Geode/utils/web.hpp (hunk 3 above)
Comment (SpaghettDev) [reply loader/include/Geode/utils/web.hpp:145]: there is no followRedirects lol, it used to be default enabled, this PR made it a default false option. I can rename it to `followRedirects` if you want

Comment (matcool) [reply loader/include/Geode/utils/web.hpp:145]: it now defaults to not following redirects? i feel like that is a bad idea, subtle api change :nedfast:

Comment (SpaghettDev) [reply loader/include/Geode/utils/web.hpp:145]: ? https://github.com/SpaghettDev/geode/blob/b8509107a971c31a47babe337c8cb83ad29e8083/loader/src/utils/web.cpp#L202

Comment (matcool) [reply loader/include/Geode/utils/web.hpp:145]: yes honey, it used to default to true and now, as you said, it defaults to false

Comment (SpaghettDev) [reply loader/include/Geode/utils/web.hpp:145]: mb that was a mistake on my part, i meant i made it a default-true option

**Review by matcool — APPROVED** [review]

**Comment by matcool** [comment]
@Cvolton any thoughts

**Review by Cvolton — APPROVED** [review]
lgtm

**Pull request merged**